### PR TITLE
Add texture type and return type tests to

### PR DESCRIPTION
**Pull request by greggman**
textureGather and textureGatherCompare

Also, added missing types to existing tests.

Issue: https://github.com/gpuweb/cts/issues/3586 https://github.com/gpuweb/cts/issues/3594

<hr>

**Requirements for PR author:**

- [X] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [X] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [X] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [X] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by dj2** [comment]
Is `src/resources/cache/webgpu/shader/execution/bitcast.bin` related to this cl  or leftover from something else?

File: src/webgpu/util/conversion.ts [hunk 1]
```diff
@@ -1004,10 +985,9 @@ export const Type = {
   vec4f: VectorType.create(4, f32Type),
   vec4h: VectorType.create(4, f16Type),
   vec4b: VectorType.create(4, boolType),
+} as const;
 
-  mat: (cols: number, rows: number, elementType: ScalarType) =>
-    MatrixType.create(cols, rows, elementType),
-
+const kMatTypes = {
```
Comment (dj2) [inline src/webgpu/util/conversion.ts:990]: Is this the same as `kAllMatrices` on line 2566?

File: src/webgpu/util/conversion.ts [hunk 2]
```diff
@@ -962,27 +963,7 @@ const boolType = new ScalarType('bool', 4, false, (buf: Uint8Array, offset: numb
...
-  'abstract-float': abstractFloatType,
-  f64: f64Type,
-  f32: f32Type,
-  f16: f16Type,
-
-  bool: boolType,
-
-  vec: (width: number, elementType: ScalarType) => VectorType.create(width, elementType),
-
+const kVecTypes = {
```
Comment (dj2) [inline src/webgpu/util/conversion.ts:966]: nit: kAllVecTypes

File: src/webgpu/util/conversion.ts (hunk 1 above)
Comment (greggman) [reply src/webgpu/util/conversion.ts:990]: No, `kMatTypes` is a record of types by name, `kAllMatrices` as an array. But kAllMatrices can be generated from `kMatTypes` so made that change

File: src/webgpu/util/conversion.ts (hunk 2 above)
Comment (greggman) [reply src/webgpu/util/conversion.ts:966]: Similar to `kAllMatrices`, generated a `kAllVecTypes` rather than change this.

**Review by dj2 — APPROVED** [review]

**Comment by greggman** [comment]
> is src/resources/cache/webgpu/shader/execution/bitcast.bin related to this cl or leftover from something else?

It's what happens when I run `npm run test`. If I don't check it in then github complains that it changes when it does its own `npm run test` and finds that files changed.

This kind of thing happens all the time to me and is one of the reasons people complain that the .bin files and hashes.json need to be generated github actions and checked into a different branch and have dawn pull that branch instead of having them in `main` and always causing these issues.

**Pull request merged**